src/main/java/frc/robot/commands/vision/CalculateStdDevs.java

```diff
@@ -0,0 +1,139 @@
+   * @param drive The drivetrain
+   * @param vision The vision
+   */
+  public CalculateStdDevs(int time, ShuffleboardTab shuffleboardTab, Drivetrain drive, Vision vision) {
+    m_tab = shuffleboardTab;
+    m_drive = drive;
+    m_vision = vision;
+    m_arrayLength = time;
+  }
+
+  /**
+   * Resets the pose array
+   */
+  @Override
+  public void initialize() {
+    m_poses = new ArrayList<Pose2d>();
+  }
+
+  /**
+   * Adds a pose to the array
+   */
+  @Override
+  public void execute() {
+    Pose2d pose = m_vision.getPose2d(m_drive.getPose());
+    // If the pose exists, add it to the first open spot in the array
+    if (pose != null) {
+      m_endTimer.stop();
+      m_endTimer.reset();
+      m_poses.add(pose);
+      System.out.printf("%.1f%% done", ((double)m_poses.size())/m_arrayLength * 100);
+    } else {
+      m_endTimer.start();
+      // If 5 seconds have passed since it saw an April tag, stop the command
+      // Prevents it from running forever
+      if(m_endTimer.hasElapsed(5)){
+        cancel();
+      }
+    }
+  }
+
+  /**
+   * Calculates the standard deviation
+   */
+  @Override
+  public void end(boolean interrupted) {
+    // If the array is empty, don't try to calculate std devs
+    if(m_poses.size() == 0){
+      System.out.println("There are no poses in the array\nTry again where the robot can see an April tag.");
+      return;
+    }
+    
+    // Calculate std devs
+    double meanX = 0;
+    double meanY = 0;
+    double meanRot = 0;
+    for (int i = 0; i < m_poses.size(); i++) {
+      meanX += m_poses.get(i).getX();
+      meanY += m_poses.get(i).getY();
+      meanRot += m_poses.get(i).getRotation().getRadians();
+    }
+    meanX /= m_poses.size();
+    meanY /= m_poses.size();
+    meanRot /= m_poses.size();
+    double totalX = 0;
+    double totalY = 0;
+    double totalRot = 0;
+    for (int i = 0; i < m_poses.size(); i++) {
+      totalX += Math.pow(m_poses.get(i).getX() - meanX, 2);
+      totalY += Math.pow(m_poses.get(i).getY() - meanY, 2);
+      totalRot += Math.pow(m_poses.get(i).getRotation().getRadians() - meanRot, 2);
+    }
+    double stdDevX = Math.sqrt(totalX / m_poses.size());
+    double stdDevY = Math.sqrt(totalY / m_poses.size());
+    double stdDevRot = Math.sqrt(totalRot / m_poses.size());
+    
+    // Calculate distance to closest April tag
+    double closest = Double.POSITIVE_INFINITY;
+    ArrayList<EstimatedRobotPose> estimatedPoses = m_vision.getEstimatedPoses(m_drive.getPose());
+    for (int i = 0; i < estimatedPoses.size(); i++) {
+      for (int j = 0; j < estimatedPoses.get(i).targetsUsed.size(); j++) {
+        double distance = estimatedPoses.get(i).targetsUsed.get(j).getBestCameraToTarget()
+        .getTranslation().toTranslation2d().getNorm();
+        closest = Math.min(closest, distance);
+      }
+    }
+    
+    // Store data in an array to make logging and shuffleboard easier
+    double[] stdDevs = {stdDevX, stdDevY, stdDevRot, closest};
```

Comment:
First off it is not easier to log and dissplay on shuffleboard.  second the closest distance is not even a standard deviation so it shouldn't be in avariable called stdDev.

Reply:
It might be easier on some computers where printing doesn't work correctly on the rio log on VSCode, like mine. On DriverStation, it sometimes covers important prints with warnings and other prints, and I don't want to accidentally lose the data. Also, the distance is related to the std dev test, in case it changes at different distances.

Reply:
Also, when this comment means is that the array makes it easier to log and use Shuffleboard instead of creating it in the same line.